tests/test_on_data/test_gin_ecephys.py

```diff
@@ -235,34 +235,50 @@ class TestConverter(NWBConverter):
     )
     def test_neuroscope_gains(self, name, conversion_options):
         input_gain = 2.0
-        data_interface = NeuroscopeRecordingInterface
         interface_kwargs = dict(file_path=str(DATA_PATH / "neuroscope" / "test1" / "test1.dat"), gain=input_gain)
 
-        nwbfile_path = str(self.savedir / f"{data_interface.__name__}-{name}.nwb")
+        nwbfile_path = str(self.savedir / f"test_neuroscope_gains_{name}.nwb")
 
         class TestConverter(NWBConverter):
-            data_interface_classes = dict(TestRecording=data_interface)
+            data_interface_classes = dict(TestRecording=NeuroscopeRecordingInterface)
 
         converter = TestConverter(source_data=dict(TestRecording=interface_kwargs))
         converter.run_conversion(nwbfile_path=nwbfile_path, overwrite=True, conversion_options=conversion_options)
 
-        # nwb file check-test
-        io = NWBHDF5IO(nwbfile_path, "r")
-        nwbfile_in = io.read()
-        output_conversion = nwbfile_in.acquisition["ElectricalSeries_raw"].conversion
-        output_gain = output_conversion * 1e6
-        assert input_gain == pytest.approx(output_gain)
+        with NWBHDF5IO(path=nwbfile_path, mode="r") as io:
+            nwbfile_in = io.read()
+            output_conversion = nwbfile_in.acquisition["ElectricalSeries_raw"].conversion
+            output_gain = output_conversion * 1e6
+            self.assertEqual(first=input_gain, second=output_gain)
```

Comment:
@h-mayorquin When inside a `unittest.TestCase`, it's best to use the large collection of `self.assert{do_something}` functions that are available. Only some array based ones have to be done through `numpy.testing`, like below

Reply:
Yeah there's many different devOps packages with their own ways of doing things. If you're in a testing file that is using `pytest` functions (no `unittest`) then direct `assert` is the right way to go, or any of the other package-specific methods.

But here, since we're already inside one, we might as well use things like `self.assertEqual(first=object_1, second=object_2)` instead of `assert object_1==object_2`. For simple objects this difference is minimal but for more complicated objects there are a host of other `self.assert{something}` functions specific to handling those cases. Examples: lists, dictionary subsets, error catching, etc.

Reply:
I have this idea that floats should not be compared with stric equality for floating point representation issues.